src/Bootstrappers/PrefixCacheTenancyBootstrapper.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Cache\Repository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Support\Facades\Cache;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class PrefixCacheTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected null|string $originalPrefix = null;
+    protected string $storeName;
+
+    public function __construct(
+        protected Application $app
+    ) {
+    }
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        $this->originalPrefix = $this->app['config']['cache.prefix'];
+        $this->storeName = $this->app['config']['cache.default'];
+
+        $this->setCachePrefix($this->app['config']['tenancy.cache.prefix_base'] . $tenant->id);
+    }
+
+    public function revert(): void
+    {
+        $this->setCachePrefix($this->originalPrefix);
+        $this->originalPrefix = null;
```

Comment:
What's the reason for clearing the `originalPrefix`? I thought I asked this in a review already but seems like I didn't.

Reply:
I can see how that makes sense here but would like to read your reasoning first in case you have other things in mind as well. These things can be a bit tricky in bootstrappers so they need to be done carefully.

Reply:
> What's the reason for clearing the originalPrefix?

There is no specific reason. Tests are passing without this. We always set the prefix in bootstrap so clearing here doesn't need. But it also makes sense to clean it after it is being used.

Reply:
TBH, I added it because I faced failures in tests where bootstrapper used the previous tenant prefix. But the actual issue was bootstrapper was not registering as a singleton. So I register it the same in TestCase (same as RedisTenanc...). And I left this line as an extra precautionary step.